stage1/init/common/seccomp.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//+build linux
+
+package common
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	stage1commontypes "github.com/coreos/rkt/stage1/common/types"
+
+	"github.com/appc/spec/schema/types"
+	"github.com/coreos/go-systemd/unit"
+)
+
+var (
+	ErrTooManySeccompIsolators = errors.New("too many seccomp isolators specified")
+)
+
+// Systemd filter mode, see
+// https://www.freedesktop.org/software/systemd/man/systemd.exec.html#SystemCallFilter=
+const (
+	sdBlacklistMode = "~"
+	sdWhitelistMode = ""
+)
+
+// getSeccompFilter gets an appc seccomp set and an optional error mode,
+// returning those values in a format suitable for systemd consumption.
+func getSeccompFilter(opts []*unit.UnitOption, p *stage1commontypes.Pod, uid int, isolators types.Isolators) ([]*unit.UnitOption, bool, error) {
+	var filterMode string
+	flag := ""
+	seccompIsolators := 0
+	seccompErrno := ""
+	noNewPrivs := false
+	var err error
+	var seccompSet []string
+	for _, i := range isolators {
+		if seccomp, ok := i.Value().(types.LinuxSeccompSet); ok {
+			seccompIsolators++
+			// By appc spec, only one seccomp isolator per app is allowed
+			if seccompIsolators > 1 {
+				return nil, false, ErrTooManySeccompIsolators
+			}
+			switch i.Name {
+			case types.LinuxSeccompRemoveSetName:
+				filterMode = sdBlacklistMode
+				seccompSet, flag, err = parseLinuxSeccompSet(p, seccomp)
+				if err != nil {
+					return nil, false, err
+				}
+				if flag == "empty" {
+					// Opt-in to rkt default whitelist
+					seccompSet = nil
+					break
+				}
+			case types.LinuxSeccompRetainSetName:
+				filterMode = sdWhitelistMode
+				seccompSet, flag, err = parseLinuxSeccompSet(p, seccomp)
+				if err != nil {
+					return nil, false, err
+				}
+				if flag == "all" {
+					// Opt-out seccomp filtering
+					return opts, false, nil
+				}
+			}
+			seccompErrno = string(seccomp.Errno())
+		}
+	}
+
+	// If unset, use rkt default whitelist
+	if len(seccompSet) == 0 {
+		filterMode = sdWhitelistMode
+		seccompSet = RktDefaultSeccompWhitelist
+	}
+
+	// Append computed options
+	if seccompErrno != "" {
+		opts = append(opts, unit.NewUnitOption("Service", "SystemCallErrorNumber", seccompErrno))
+	}
+	seccompStr := ""
+	for i, s := range seccompSet {
+		next := i + 1
+		seccompStr += (s + " ")
+		if next%50 == 0 || next == len(seccompSet) {
```

Comment:
Uhm, you are right, I should better clarify this. The default list is pretty big, and from here till the final unit serialization it was hitting a truncation limit somewhere across the stack. Splitting them in batch here is kind of a workaround to that, but probably also makes it easier to read (50 is a completely arbitrary number). I agree this is a "meh" snippet of code. What should we do, track down the truncation further and remove it? (I fear we will anyway hit some smaller buffer)